### PR TITLE
Give `gds.run_cypher` an optional `database` parameter

### PR DESCRIPTION
**Pull request by adamnsch**

### CODE REVIEW

File: graphdatascience/tests/integration/test_database_ops.py [hunk 1]
```diff
@@ -32,6 +32,24 @@ def test_switching_db(runner: Neo4jQueryRunner) -> None:
...
 
+@pytest.mark.skip_on_aura
+def test_run_query_with_db(runner: Neo4jQueryRunner) -> None:
+    runner.run_query("CREATE (a: Node)")
+
+    default_db_count = runner.run_query("MATCH (n: Node) RETURN COUNT(n) AS c")["c"][0]
+    assert default_db_count == 1
+
+    MY_DB_NAME = "my-db"
+    runner.run_query("CREATE DATABASE $dbName", {"dbName": MY_DB_NAME})
```
Comment (FlorentinD) [inline graphdatascience/tests/integration/test_database_ops.py:43]: this is an enterprise feature (https://neo4j.com/docs/cypher-manual/current/databases/#administration-databases-create-database). So make sure we use a Neo4j enterprise edition

Comment (adamnsch) [reply graphdatascience/tests/integration/test_database_ops.py:43]: THis is a problem in other client tests as well. For now we should say that Neo4j enterprise is needed in our dev docs?

File: graphdatascience/tests/integration/test_database_ops.py [hunk 2]
```diff
@@ -32,6 +32,24 @@ def test_switching_db(runner: Neo4jQueryRunner) -> None:
...
+    # runner.set_database(MY_DB_NAME)
```
Comment (Mats-SX) [inline graphdatascience/tests/integration/test_database_ops.py:44]: should remove this comment perhaps?

File: graphdatascience/tests/integration/test_database_ops.py [hunk 3]
```diff
@@ -32,6 +32,24 @@ def test_switching_db(runner: Neo4jQueryRunner) -> None:
...
+
+    specified_db_count = runner.run_query("MATCH (n: Node) RETURN COUNT(n) AS c", database=MY_DB_NAME)["c"][0]
+    assert specified_db_count == 0
```
Comment (Mats-SX) [inline graphdatascience/tests/integration/test_database_ops.py:47]: neat test

File: graphdatascience/tests/integration/test_database_ops.py (hunk 1 above)
Comment (Mats-SX) [reply graphdatascience/tests/integration/test_database_ops.py:43]: the alternative would be to mark tests with another "neo4j-enterprise" marker. Seems a bit complicated? Maybe we keep it like this until someone complains and do the work to have a "neo4j CE"-mode later?

**Review by Mats-SX — APPROVED** [review]
Nice and simple

File: graphdatascience/tests/integration/test_database_ops.py (hunk 1 above)
Comment (FlorentinD) [reply graphdatascience/tests/integration/test_database_ops.py:43]: agreed. We also learned yesterday that Neo4j desktop includes neo4j EE by default

File: graphdatascience/query_runner/neo4j_query_runner.py [hunk 4]
```diff
@@ -61,21 +70,23 @@ def run_query_with_logging(self, query: str, params: Optional[Dict[str, Any]] =
...
-            self._log(job_id, future)
+            self._log(job_id, future, database)
 
             if future.exception():
                 raise future.exception()  # type: ignore
             else:
                 return future.result()
 
-    def _log(self, job_id: str, future: "Future[Any]") -> None:
+    def _log(self, job_id: str, future: "Future[Any]", database: Optional[str] = None) -> None:
```
[the remark this answers is not in the document]
Comment (FlorentinD) [reply graphdatascience/query_runner/neo4j_query_runner.py:82]: We could but I dont see much value in skipping it here.

**Pull request merged**